packages/core/src/State.ts

```diff
@@ -63,7 +69,14 @@ export class State<
   /**
    * An object mapping actor names to spawned/invoked actors.
    */
-  public children: Record<string, ActorRef<any>>;
+  public children: TActor['id'] extends string
```

Comment:
TODO: this could be improved. The fallback branch here kicks in way too early. For example: when we **provide** `TActor` without any IDs then we should create an index signature of all provided actor types. In fact, this should also happen for a "mixed" bag of actors (when some have IDs and when some don't). In other words, the index signature containing possible actor types should always be available (unless all actors have explicit IDs assigned to them?).

Reply:
fixed https://github.com/statelyai/xstate/pull/4036/commits/041b40b8bcddbec420471855f7cd5b1bc47c64d2 , https://github.com/statelyai/xstate/pull/4036/commits/2e7a58bc1428185ebd522b3d7abbf3739d267db6